src/utils.ts

```diff
@@ -208,23 +209,38 @@ export async function getListOfTargets(hostname: string, port: number, useHttps:
 
     const protocol = (useHttps ? 'https' : 'http');
 
-    let jsonResponse = '';
+    let jsonResponse = null;
     for (const endpoint of ['/json/list', '/json']) {
         try {
             jsonResponse = await checkDiscoveryEndpoint(`${protocol}://${hostname}:${port}${endpoint}`);
             if (jsonResponse) {
                 break;
             }
-        } catch {
-            // Do nothing
+        } catch (e) {
+            // localhost might not be ready as the user might not have a server running
+            // so just show an error dialog for any other condition.
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            if (e && e.code && e.code === 'ECONNREFUSED'){
+                continue;
+            }
+
+            void ErrorReporter.showErrorDialog({
```

Comment:
why do these need to be marked 'void'?

Reply:
As no response is expected and because VS will end up handling them I think is better to have a fire and forget in here.